examples/transactions.c

```diff
@@ -146,26 +121,21 @@ static void dr_msg_cb (rd_kafka_t *rk,
 
 
 /**
- * @brief Create a transactional producer for the given input pratition
- *        and begin a new transaction.
+ * @brief Create a transactional producer.
  */
 static rd_kafka_t *
-create_transactional_producer (const rd_kafka_topic_partition_t *rktpar) {
+create_transactional_producer (const char *brokers, const char *output_topic) {
         rd_kafka_conf_t *conf = rd_kafka_conf_new();
         rd_kafka_t *rk;
         char errstr[256];
         rd_kafka_error_t *error;
-        char transactional_id[256];
-
-        snprintf(transactional_id, sizeof(transactional_id),
-                 "librdkafka_transactions_example_%s-%d",
-                 rktpar->topic, rktpar->partition);
 
         if (rd_kafka_conf_set(conf, "bootstrap.servers", brokers,
                               errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK ||
-            rd_kafka_conf_set(conf, "transactional.id", transactional_id,
+            rd_kafka_conf_set(conf, "transactional.id",
+                              "librdkafka_transactions_example",
```

Comment:
maybe have a clientid command line arg and construct the transactional.id from that? else you can't run more than one instance of the example app at once to spread the load.